src/scripts/main.js

```diff
@@ -1,3 +1,73 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('.field');
+  const appendRowBtn = document.querySelector('.append-row');
+  const removeRowBtn = document.querySelector('.remove-row');
+  const appendColumnBtn = document.querySelector('.append-column');
+  const removeColumnBtn = document.querySelector('.remove-column');
+
+  const MAX_COUNT = 10;
+  const MIN_COUNT = 2;
+
+  function updateButtons() {
+    const rowCount = table.rows.length;
+    const colCount = table.rows[0]?.cells.length || 0;
```

Comment:
Ensure that the table has at least one row before accessing `table.rows[0]`. If the table is empty, this could lead to a runtime error. Consider initializing the table with the minimum number of rows and columns.